packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer-node.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+	<template v-if="model">
+		<div class="container">
+			<tera-model-diagram :model="model" :is-editable="false" nodePreview />
+		</div>
+	</template>
+</template>
+
+<script setup lang="ts">
+import { getModel } from '@/services/model';
+import { Model } from '@/types/Types';
+import { WorkflowNode } from '@/types/workflow';
+import { onMounted, ref, watch } from 'vue';
+import TeraModelDiagram from '@/components/model/petrinet/model-diagrams/tera-model-diagram.vue';
+
+const props = defineProps<{
+	node: WorkflowNode;
+}>();
+
+const model = ref<Model | null>(null);
+
+const fetchModel = async () => {
+	if (!props.node?.state?.modelId) return;
+	model.value = await getModel(props.node?.state?.modelId);
+};
+
+onMounted(async () => {
+	await fetchModel();
+});
+watch(
+	() => props.node.state,
```

Comment:
You may want to tighten this watcher a bit if the only thing you need is the modelId field,

Reply:
done!